gui/src/ScenarioSelector.tsx

```diff
@@ -9,31 +65,332 @@ interface ScenarioSelectorProps {
+    .required("Config is required."),
+  name: Yup.string()
+    .trim("Cannot include leading and trailing spaces")
+    .required("Name is required."),
+  date: Yup.string().required("Date is required."),
+});
+
 const ScenarioSelector = (props: ScenarioSelectorProps) => {
-  const { } = props;
+  const {} = props;
+
+  const [open, setOpen] = useState(false);
+  const [nodes, setNodes] = useState<TreeNode[]>([]);
 
-  const showAddButton = useDynamicProperty(props.showAddButton, props.defaultShowAddButton, true);
-  const displayCycles = useDynamicProperty(props.displayCycles, props.defaultDisplayCycles, true);
-  const showPrimaryFlag = useDynamicProperty(props.showPrimaryFlag, props.defaultShowPrimaryFlag, true);
-  const scenarioId = useDynamicProperty(props.scenarioId, props.defaultScenarioId, "");
   const dispatch = useDispatch();
   const module = useModule();
 
+  const theme = createTheme({
+    spacing: 1,
+    palette: {
+      mode: "dark",
+    },
+  });
+
+  const showAddButton = useDynamicProperty(
+    props.showAddButton,
+    props.defaultShowAddButton,
+    true
+  );
+  const displayCycles = useDynamicProperty(
+    props.displayCycles,
+    props.defaultDisplayCycles,
+    true
+  );
+  const showPrimaryFlag = useDynamicProperty(
+    props.showPrimaryFlag,
+    props.defaultShowPrimaryFlag,
+    true
+  );
+  const scenarioId = useDynamicProperty(
+    props.scenarioId,
+    props.defaultScenarioId,
+    ""
+  );
+
+  //mock data
+  const scenarios = [
+    {
+      cycle: "CYCLE_863418_fdd1499a-8925-4540-93fd-9dbfb4f0846d",
+      id: "SCENARIO_63cb358d-5834-4d73-84e4-a6343df5e08c",
+      properties: {},
+      tags: [],
+      version: "latest",
+      pipelines: [
+        "PIPELINE_mean_baseline_5af317c9-34df-48b4-8a8a-bf4007e1de99",
+        "PIPELINE_arima_90aef6b9-8922-4a0c-b625-b2c6f3d19fa4",
+      ],
+      subscribers: [],
+      creation_date: "2022-08-15T19:21:01.871587",
+      primary_scenario: true,
+    },
+    {
+      cycle: "CYCLE_863418_fdd1499a-8925-4540-93fd-9dbfb4f0846d",
+      id: "SCENARIO_2222222222263cb358d-5834-4d73-84e4-a6343df5e08c",
+      properties: {},
+      tags: [],
+      version: "latest",
+      pipelines: [
+        "PIPELINE_mean_baseline_5af317c9-34df-48b4-8a8a-bf4007e1de99",
+        "PIPELINE_arima_90aef6b9-8922-4a0c-b625-b2c6f3d19fa4",
+      ],
+      subscribers: [],
+      creation_date: "2022-08-15T19:21:01.871587",
+      primary_scenario: false,
+    },
+    {
+      cycle: "CYCLE_863418_fdd1499a-8925-4540-93fd-9dbfb4f0846d",
+      id: "SCENARIO_333333333333333363cb358d-5834-4d73-84e4-a6343df5e08c",
+      properties: {},
+      tags: [],
+      version: "latest",
+      pipelines: [
+        "PIPELINE_mean_baseline_5af317c9-34df-48b4-8a8a-bf4007e1de99",
+        "PIPELINE_arima_90aef6b9-8922-4a0c-b625-b2c6f3d19fa4",
+      ],
+      subscribers: [],
+      creation_date: "2022-08-16T19:21:01.871587",
+      primary_scenario: true,
+    },
+  ];
```

Comment:
We should extract this in another external ts file.
And we should settle on a data format. I would propose something close to LoV